### PR TITLE
Add adaptive auth concepts

### PR DESCRIPTION
**Pull request by sherenewso2**

### CODE REVIEW

File: en/docs/concepts/authentication/adaptive-authentication.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Adaptive Authentication
+
+## What is adaptive authentication?
+
+Adaptive authentication, also known as step-up authentication, is an evolved and flexible form of [multi-factor authentication](TODO:link-to-concept) that tightens security when the possibility of a security breach is higher. Additional authentication steps such as SMS OTP, FIDO, fingerprint authentication etc., can be configured and deployed in such a way so that the system is able to decide which steps to prompt during the authentication process depending on the user’s risk profile and their behaviour.  
+
+For instance, high-risk transactions such as a user attempting to tranfer a high sum of money causes the adaptive authentication mechanism to prompt an extra authentication step like sending a one time security code to the user's mobile phone, in order to increase security.
```
Comment (gomathyk) [inline en/docs/concepts/authentication/adaptive-authentication.md:7]: one-time

File: en/docs/concepts/authentication/adaptive-authentication.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    -   [Five instances to use adaptive
+        authentication](https://wso2.com/library/article/2018/10/5-instances-to-use-adaptive-authentication/)
+    -   [Four reasons to use WSO2 Identity Server for adaptive
+        authentication](https://wso2.com/library/article/2018/10/four-reasons-to-use-wso2-is-for-adaptive-authentication/)
+
+---
+
+## Using adaptive authentication
+
+WSO2 Identity Server (WSO2 IS) supports script-based adaptive authentication, which allows you to use a script to set up appropriate authentication factors depending on your scenario using the provided script editor. The script editor provides a set of predefined templates that you can use to easily set up adaptive authentication for some of the most common authentication scenarios. If necessary, you can also use the script editor to introduce new functions and fields to an authentication script based on your requirement, and then engage the script to the service provider’s authentication step configuration.
```
Comment (gomathyk) [inline en/docs/concepts/authentication/adaptive-authentication.md:43]: Maybe we can avoid using you and your?

**Review by gomathyk — CHANGES_REQUESTED** [review]

File: en/docs/concepts/authentication/adaptive-authentication.md (hunk 2 above)
Comment (sherenewso2) [reply en/docs/concepts/authentication/adaptive-authentication.md:43]: will address this in the next PR

File: en/docs/concepts/authentication/adaptive-authentication.md (hunk 1 above)
Comment (sherenewso2) [reply en/docs/concepts/authentication/adaptive-authentication.md:7]: will address this in the next PR

**Review by gomathyk — APPROVED** [review]

**Pull request merged**